src/models/vae.jl

```diff
@@ -0,0 +1,150 @@
+using Flux
+using ConditionalDists
+using GenerativeModels
+using ValueHistories
+using MLDataPattern: RandomBatches
+using Distributions
+using StatsBase
+using Random
+
+"""
+	safe_softplus(x::T)
+
+Safe version of softplus.	
+"""
+safe_softplus(x::T) where T  = softplus(x) + T(0.000001)
+
+
+"""
+	vae_constructor(;idim::Int=1, zdim::Int=1, activation = "relu", hdim=128, nlayers=3, kwargs...)
+"""
+function vae_constructor(;idim::Int=1, zdim::Int=1, activation = "relu", hdim=128, nlayers::Int=3, init_seed::Int=nothing, kwargs...)
+	(nlayers < 2) ? error("Less than 3 layers are not supported") : nothing
+	# function from string
+	act = eval(Meta.parse(activation))
+
+	# if seed is given, set it
+	(init_seed != nothing) ? Random.seed!(init_seed) : nothing
+	
+	# construct the model
+	# encoder - diagonal covariance
+	encoder_map = Chain(
```

Comment:
I see that you did not use the `build_mlp` methods. Is there something missing in their implementation?

Reply:
```julia
decoder_map = Chain(
		build_mlp(idim, hdim, hdim, nlayers-1, activation=activation, lastlayer="linear")...,
		ConditionalDists.SplitLayer(hdim, [zdim, zdim], [identity, safe_softplus])
		)
```
Should be equivalent, but I have not checked it.

Reply:
I wanted to use it but did not get to it - thanks!